MODAK/src/MODAK.py

```diff
@@ -111,6 +123,45 @@ def get_opt_container_runtime(self, job_json_data):
             new_container = self._map.map_container(job_json_data)
         logging.info(f"Optimal container: {new_container}")
         return new_container if new_container else ""
+    
+    def get_buildjob(self, job_json_data):
+        logging.info('Creating build script for job')
+        logging.info('Processing build data' + str(job_json_data))
+
+        # patch in the build job as the job task
+        # First, figure out what we're patching in:
+        try:
+            buildsrc = job_json_data["job"]["application"]["build"]["src"]
+        except KeyError:
+            logging.info("No job.application.build.src section, returning empty response")
+            return ""
+        buildcmd = job_json_data["job"]["application"]["build"]["build_command"]
+        final_build = ""
+        if buildsrc[-4:] == ".git":
+            final_build = f"git clone {buildsrc}\n"
+        else:
+            final_build = f"wget --no-check-certificate {buildsrc}\n"
+        
+        process_per_node=1
+        try:
+            process_per_node = job_json_data["job"]["application"]["build"]["build_parallelism"]
+        except KeyError:
+            pass
+        final_build += re.sub(
+            pattern="\{\{[\s]*BUILD_PARALLELISM[\s]*\}\}",
+            repl=str(process_per_node),
+            string=buildcmd
+        )
+
+        mod_jobdata = deepcopy(job_json_data)
+        mod_jobdata["job"]["job_options"]["job_name"] = job_json_data["job"]["job_options"]["job_name"] + "_build"
+        mod_jobdata["job"]["job_options"]["node_count"] = 1
+        mod_jobdata["job"]["job_options"]["process_count_per_node"] = process_per_node
+        mod_jobdata["job"]["job_options"]["standard_output_file"] = "build-" + job_json_data["job"]["job_options"]["standard_output_file"]
+        mod_jobdata["job"]["job_options"]["standard_error_file"] = "build-" + job_json_data["job"]["job_options"]["standard_error_file"]
```

Comment:
```suggestion
        build_job_options = mod_jobdata["job"]["job_options"]
        build_job_options.update({
            "job_name": f"{build_job_options['job_name']}_build",
            "node_count": 1,
            "process_count_per_node": process_per_node,
            "standard_output_file"] = f"build-{build_job_options['standard_output_file']}",
            "standard_error_file"] = f"build-{build_job_options['standard_error_file']}",
        })
```

Reply:
I personally don't like the .update() syntax of dictionaries. While it doesn't hide what's going on, it's uncommon (in my experience) and slightly un-pythonic. (It seems more like a Java convention to me). Repetitive as it is, repeated assignments directly to a key are more clear about what's going on.

I can change this, of course, if you'd still prefer the .update() syntax.

I will update this to use format strings.

Reply:
> I personally don't like the .update() syntax of dictionaries.

Me neither, but it seems to be a common pattern in MODAK so I stayed with it for now, but I don't mind if we get rid of it.